Client/Frontend/Browser/GridTabViewController.swift

```diff
@@ -384,7 +384,7 @@ extension GridTabViewController {
         } else if self.tabManager.normalTabs.count == 1, let tab = self.tabManager.normalTabs.first {
             self.tabManager.selectTab(tab)
             self.dismissTabTray()
-            notificationCenter.post(name: .TabsTrayDidClose)
+            notificationCenter.post(name: .TabsTrayDidClose, anObject: nil)
```

Comment:
We can have a default parameter on the protocol so we don't have to specify that the object is nil here (same comment applies for your changes in the files `TabTrayViewController`, `HomePageViewController`, `SearchSettingsTableViewController`, etc. We can set default parameter on protocol by adding an extension to the said protocol.

```
extension NotificationProtocol {
    func post(name: NSNotification.Name, anObject: Any? = nil) { 
        self.post(name: name, anObject: anObject)
    }
}
```

Reply:
After making this change and removing `anObject: nil` from `notificationCenter.post() ` in `GridTabViewController`, I was still getting these errors:

<img width="1280" alt="Screenshot 2023-03-27 at 11 36 28 PM" src="https://user-images.githubusercontent.com/114816569/228030066-7092c518-2d1b-42a5-9732-8278e7218318.png">

<img width="318" alt="Screenshot 2023-03-24 at 7 22 04 PM" src="https://user-images.githubusercontent.com/114816569/228030753-671b0f3f-14cf-4390-b07f-c0875edd2bd4.png">

Reply:
It needs to be in an extension on the protocol, not in the `NotificationCenter: NotificationProtocol`. You want to extend the functionnality of the protocol and not just the NotificationCenter object. See my previous comment.

Reply:
Yes, from what I see that should work.

Reply:
Still getting same error: 

<img width="1237" alt="Screenshot 2023-03-28 at 10 00 55 PM" src="https://user-images.githubusercontent.com/114816569/228307620-83995b82-0ac7-43bf-b832-3303b6df983a.png">

Reply:
So the Notifable protocol is in the BrowserKit package, you need to make sure the extension is public so the rest of the project can access it.